base_decorator.rb

```diff
@@ -0,0 +1,12 @@
+require_relative 'nameable'
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    super
```

Comment:
- [ ] I have ran your code and it gives error. but if you remove `super` keyword from line 5 and add `attr_accessor :nameable` right after line 3, then you code will be working. kindly consider fixing the bug.

Reply:
Hello @Zuhaib042,
The bug is resolved, thanks for your insight, much appreciates.